### DISCUSSION TITLE
Can I clear the Results pane in code?

### DISCUSSION THREAD

**Discussion by jimkilliansentinel** [microsoft/azuredatastudio#21923] [Q&A]
I'm running this code to check some queues on a schedule and would like to clear the Results pane before each iteration so that the current results are always in view:
```
    while ({some test}) 
    BEGIN  
        {clear Results pane}  
        exec [dbo].[_JimK_CheckJobQueues];  
        RAISERROR( 'Just to print everything',0,1) WITH NOWAIT  
        WAITFOR DELAY @waitTime;  
    END  
```
The code works fine except for clearing the Results pane. I had to add the RAISERROR to get the results to show before pausing.

Is this doable?

**Comment by EmergeMarketing** [comment]
I don't believe there's a built-in command to clear the Results pane directly from a T-SQL script running in ADS. The Results pane is managed by the user interface of the tool, rather than the T-SQL script itself.

If you need to clear the Results pane frequently, you can click the "Clear Result" button. However, your specific need seems to be having each iteration's results displayed distinctly. One workaround for this is to add a PRINT or RAISERROR statement at the start of each iteration to indicate a new set of results. This would make it easier to visually separate the results of each iteration.

However, Azure Data Studio is related to VS Code and features added to VS Code often end up being supported in Azure Data Studio. There is an issue raised for this feature in the [VS Code repository](https://github.com/microsoft/vscode/issues/10121), and it is currently in the backlog.

Hopefully that helps.

Jack @ [Emerge](https://emerge.digital)

**Comment by jimkilliansentinel** [comment]
Thanks, Jack!

I figured that T-SQL wouldn’t be able to directly affect the ADS UI, but still hoped that there might be some kind of directive that could be put in the code (maybe as a comment) that ADS would be able to intercept and act upon.

Maybe some enterprising user (who’s a better programmer than I am) will find a way to do this in an extension?

Jim

Jim Killian | Software Developer
Sentinel Offender Services, LLC
111 Corporate Drive, Suite 125 | Ladera Ranch, CA 92694<webextlink://111%20Corporate%20Drive,%20Suite%20125%20|%20Ladera%20Ranch,%20CA%2092694/>
C| 949.239.3722
E| ***@***.******@***.***>
W| www.sentineladvantage.com<http://www.sentineladvantage.com/>
